src/OrbitGl/App.h

```diff
@@ -627,10 +627,27 @@ class OrbitApp final : public DataViewFactory,
     orbit_base::StopSource stop_source;
     orbit_base::Future<ErrorMessageOr<std::filesystem::path>> future;
   };
-  // ONLY access this from the main thread
-  absl::flat_hash_map<std::pair<std::string, std::string>, ModuleLoadOperation>
-      modules_currently_loading_;
-  // ONLY access this from the main thread
+  // Map of module file path to download operation future, that holds all symbol downloads that
+  // are currently in progress.
+  // ONLY access this from the main thread.
+  absl::flat_hash_map<std::string, ModuleLoadOperation> symbol_files_currently_downloading_;
+
+  // Map of "module ID" (file path and build ID) to symbol file retrieving future, that holds all
+  // symbol retrieving operations currently in progress. (Retrieving here means finding locally or
+  // downloading from the instance). Since downloading a symbols file can be part of the retrieval,
+  // if a module ID is contained in symbol_files_currently_downloading_, it is also contianed in
+  // symbol_files_currently_retrieved_.
+  // ONLY access this from the main thread.
+  absl::flat_hash_map<std::pair<std::string, std::string>,
+                      orbit_base::Future<ErrorMessageOr<std::filesystem::path>>>
+      symbol_files_currently_retrieved_;
```

Comment:
I'm not sure about the name "currently_retrieved", it feels like if a file is retrieved, the work has been done and there is nothing left to do. I would expect a file to be "retrieved" once downloading is finished, not while it's being downloaded. Maybe "symbol_files_currently_being_retrieved" ?

Reply:
+1

Reply:
Good idea! Done